product_import/wizard/product_import.py

```diff
@@ -213,33 +248,33 @@ def create_product(self, parsed_product, chatter_msg, seller=None):
                 # all characteristics into product.template
                 product.flush()
                 product.action_archive()
-            logger.info("Product %d created", product.id)
-        return product
+            logger.debug("Product %s created", product.default_code)
 
-    @api.model
-    def _create_products(self, catalogue, seller, filename=None):
-        products = self.env["product.product"].browse()
-        for product in catalogue.get("products"):
-            record = self.create_product(
-                product,
-                catalogue["chatter_msg"],
-                seller=seller,
-            )
-            if record:
-                products |= record
-        self._bdimport.post_create_or_update(catalogue, seller, doc_filename=filename)
-        logger.info("Products updated for vendor %d", seller.id)
-        return products
+        log_msg = f"Product created/updated {product.id}\n" + "\n".join(chatter_msg)
+        return log_msg
 
-    def import_button(self):
+    def import_button(self, chunk_size=CHUNK_SIZE):
         self.ensure_one()
         file_content = b64decode(self.product_file)
         catalogue = self.parse_product_catalogue(file_content, self.product_filename)
         if not catalogue.get("products"):
             raise UserError(_("This catalogue doesn't have any product!"))
         company_id = self._get_company_id(catalogue)
         seller = self._get_seller(catalogue)
-        self.with_context(product_company_id=company_id)._create_products(
-            catalogue, seller, filename=self.product_filename
+        wiz = self.with_context(product_company_id=company_id)
+        # Create products asynchronously
+        iterators = [iter(catalogue["products"])] * chunk_size
+        for products in zip_longest(*iterators):
+            if products[-1] is None:
+                products = [product for product in products if product]
+            # One job for x products (chunk of 40)
+            wiz.with_delay()._create_update_products(products, seller.id)
```

Comment:
here I would simply call `wiz.create_update_products` that will call `_create_update_products` which can be overridden or you can rely on https://github.com/OCA/queue/blob/16.0/queue_job/models/base.py#L145